lib/ansible/module_utils/common/validator.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020 Ansible Project
+# Simplified BSD License (see licenses/simplified_bsd.txt or https://opensource.org/licenses/BSD-2-Clause)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+from ansible.module_utils.common._collections_compat import (
+    Sequence,
+)
+
+from ansible.module_utils.common.parameters import (
+    get_unsupported_parameters,
+    validate_argument_types,  # Rename this because it actually does coercion?
+    validate_argument_values,
+)
+
+from ansible.module_utils.common.text.converters import to_native
+from ansible.module_utils.common.validation import (
+    check_required_arguments,
+)
+
+from ansible.module_utils.six import string_types
+
+
+class Validator():
+    """Argument spec validator class"""
+
+    def __init__(self):
+        self._options_context = None
+
+        self._error_messages = []
+        self._validated_parameters = {}
+
+    @property
+    def error_messages(self):
+        return self._error_messages
+
+    @property
+    def validated_parameters(self):
+        return self._validated_parameters
+
+    def add_error(self, error):
```

Comment:
Since we don't intend for a user of this class to call this method, let's make it "private" by renaming with a leading underscore.